### ISSUE TITLE
Bug: Suspect multiple prices for same token

### ISSUE THREAD

**Issue by bh2smith** [cowprotocol/solver-rewards#29]
Based on this conversation in slack:

https://cowservices.slack.com/archives/C0361CCTFD5/p1653749523942819

Our slippage query is WAY off for this transaction `0xd34d62c05ce53573313c47285f4d045a3e60e12fd658142abc27f03eac0fb02d`

Due to the Flora Token. 

My hunch is that the extreme positive slippage could be from two different prices for the one token.

**Comment by bh2smith** [comment]
I have started digging into some guesses here,

- [x] This token is not in `prices.usd`
- [x] we do have consistent hourly median prices available for this token at the time of trade from `prices.prices_from_dex_data`
- [x] Also this is definitely not an issue with multiplicity in clearing prices.

all of the above facts can be verified with [this query](https://dune.com/queries/861915) and commented blocks

Next to look into is possible numerical overflow:

**Comment by josojo** [comment]
I think the issue is that the token price is ridiculous high! Due to that, a tiny slippage was evaluated very nicely for the solver.
![image](https://user-images.githubusercontent.com/7348154/170971565-36242556-99cc-4eb4-a970-004a56371309.png)

**Comment by josojo** [comment]
If we are changing the endPrices to only be considered, if there is a bigger sample size, we avoid this error:
```
end_prices as (
    select
        median_price as price,
        p_complete.contract_address,
        decimals
    from
        prices.prices_from_dex_data p_complete
    where
        p_complete.hour = '{{EndTime}}'
        and sample_size >= '3'
),
```
```
Adjusting 0x080a8b1E2f3695E179453c5e617b72A381BE44B9(barn-ParaSwap) transfer by -0.00170 (slippage)
Redirecting solver 0x080a8b1E2f3695E179453c5e617b72A381BE44B9 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0x109BF9E0287Cc95cc623FBE7380dD841d4bdEb03(barn-Otex) transfer by -0.20368 (slippage)
Redirecting solver 0x109BF9E0287Cc95cc623FBE7380dD841d4bdEb03 COW tokens to 0x84dbAE2549d67CaF00f65C355de3D6F4df59A32C
Adjusting 0x15F4c337122Ec23859EC73Bec00aB38445E45304(prod-ParaSwap) transfer by -0.27001 (slippage)
Redirecting solver 0x15F4c337122Ec23859EC73Bec00aB38445E45304 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0x2d15894FaC906386fF7f4Bd07fcEAc43fcF80C73 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0x340185114F9D2617DC4c16088d0375D09fee9186(prod-Naive) transfer by -0.00139 (slippage)
Redirecting solver 0x340185114F9D2617DC4c16088d0375D09fee9186 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0x6fa201C3Aff9f1e4897Ed14c7326cF27548d9c35(prod-Otex) transfer by -6.69399 (slippage)
Redirecting solver 0x6fa201C3Aff9f1e4897Ed14c7326cF27548d9c35 COW tokens to 0x84dbAE2549d67CaF00f65C355de3D6F4df59A32C
Adjusting 0x77ec2A722c2393D3fD64617BBaF1499C713e616b(prod-QuasiModo) transfer by -0.63251 (slippage)
Redirecting solver 0x77ec2A722c2393D3fD64617BBaF1499C713e616b COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0x833f076D182123cA8DDE2743045Ea02957bD61ef COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0xa0044c620Da7f2876dA7004719b8370eb7BE5e50 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0xa97DEF4fBCBa3b646dd169Bc2eeE40f0f3fE7771(prod-BalancerSOR) transfer by -0.00387 (slippage)
Redirecting solver 0xa97DEF4fBCBa3b646dd169Bc2eeE40f0f3fE7771 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0xda324c2f06d3544E7965767ce9ca536dcb67a660 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0xdAE69AffE582d36f330Ee1145995A53Fab670962 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0xdE1c59Bc25D806aD9DdCbe246c4B5e5505645718(prod-1inch) transfer by -0.57735 (slippage)
Redirecting solver 0xdE1c59Bc25D806aD9DdCbe246c4B5e5505645718 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0xDe786877a10DBb7EBa25a4DA65aEcf47654F08ab(barn-0x) transfer by -0.00295 (slippage)
Redirecting solver 0xDe786877a10DBb7EBa25a4DA65aEcf47654F08ab COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0xe33062a24149f7801a48B2675Ed5111d3278F0f5(barn-1inch) transfer by -0.01456 (slippage)
Redirecting solver 0xe33062a24149f7801a48B2675Ed5111d3278F0f5 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0xe92F359e6F05564849AFa933CE8F62b8007A1d5D COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0xf2D21ad3c88170D4AE52bBBeBA80Cb6078D276F4 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
dumping 19 results to transfers-2022-05-17-to-2022-05-24.csv
Total ETH Funds needed: 32.816458415780446
Total COW Funds needed: 340300.0
``

**Comment by bh2smith** [comment]
Fixed by #30
